tests/convert/test_convert.py

```diff
@@ -0,0 +1,89 @@
+import shutil
+from pathlib import Path
+
+import cv2
+import numpy as np
+from pytest import approx
+
+from OTVision.convert.convert import check_ffmpeg
+from OTVision.convert.convert import main as convert
+from OTVision.helpers.files import get_files
+
+
+def test_check_ffmpeg() -> None:
+    """Tests if ffmpeg can be called as a subprocess"""
+    check_ffmpeg()
+
+
+def test_convert(test_data_tmp_dir: Path, test_data_dir: Path) -> None:
+    """Tests the main function of OTVision/convert/convert.py
+    transforming short test videos from h264 to mp4 based on
+    framerate specified as part of the file path using ffmpeg.exe
+    """
+
+    # Get reference data
+    h264_ref_videos = get_files(paths=[test_data_dir], filetypes=[".h264"])
+    mp4_ref_videos = get_files(paths=[test_data_dir], filetypes=[".mp4"])
+
+    # Copy input data to temporary folder and start conversion
+    for h264_ref_video in h264_ref_videos:
+        shutil.copy2(
+            Path(h264_ref_video),
+            Path(h264_ref_video).parents[1]
+            / test_data_tmp_dir.name
+            / Path(h264_ref_video).name,
+        )
+    h264_test_videos = get_files(
+        paths=[test_data_tmp_dir],
+        filetypes=[".h264"],
+    )
+    convert(h264_test_videos)
+
+    # Local function to turn videos into np arrays for comparison
+    def array_from_video(file: Path) -> np.ndarray:
```

Comment:
Looks like there is no reference to the outer function. Therefore, this method can be moved on top level.